### ISSUE TITLE
Proposal : make Sec-WebSocket-Protocol header optional or param that changes its value

### ISSUE THREAD

**Issue by Demayl** [croservices/cro-websocket#13]
Problem is that not all websocket servers are ignoring this header ( if unknown protocol is used ) and are returning error like
> 426 No Sec-WebSocket-Protocols requested supported

See
https://www.iana.org/assignments/websocket/websocket.xml#subprotocol-name

**Comment by maros** [comment]
Asterisk Event API gets confused by the "Sec-WebSocket-Protocol: echo" header, since it expects either 'sip' or nothing at all. However adding your own 'Sec-WebSocket-Protocol' header via
```
my $client = Cro::WebSocket::Client.new(
    uri => 'wss://...',
    headers => [
        hu
            name => 'Sec-WebSocket-Protocol',
            value => 'sip'
        )
    ]
);
```
results in having the header twice.

**Comment by jnthn** [comment]
Implemented in #29; closing.
